### PR TITLE
[WIP] [2.6] Alpine linux compatibility

### PR DESCRIPTION
**Pull request by Soullivaneuh**
Replaces #1058 #1059.

I add a docker files to test. Will remove it at the end.

To test it, install [Docker](https://www.docker.com/) and run:

``` bash
docker build -t mongo:2.6-build .
docker run --rm -it mongo:2.6-build
```

Ill update the main message with issues and question to resume the progress.

Tell me if you want another PR for 3.x or if you will backport it yourself.

Please not that Alpine Linux use [Musl](http://www.musl-libc.org/) C library.
- [x] Fix `gnu_get_libc_version` usage that does not exist on Musl
- [x] Add `/etc/alpine-release` file map
- [x] `mallinfo (void)` Does not exists on Musl. Find a generic way.
  - http://stackoverflow.com/q/34905260/1731473
- [x] Fix `'pthread_yield' was not declared in this scope`
- [x] `Dl_info_t` missing
- [ ] Fix `'walkcontext' was not declared in this scope`

Don't hesitate to help if you have some suggestion about unfinished tasks! :+1:

### CODE REVIEW

File: src/mongo/util/processinfo_linux2.cpp [hunk 1]
```diff
@@ -384,8 +391,8 @@ namespace mongo {
 
     void ProcessInfo::getExtraInfo( BSONObjBuilder& info ) {
         // [dm] i don't think mallinfo works. (64 bit.)  ??
-        struct mallinfo malloc_info = mallinfo(); // structure has same name as function that returns it. (see malloc.h)
-        info.append("heap_usage_bytes", malloc_info.uordblks/*main arena*/ + malloc_info.hblkhd/*mmap blocks*/);
+//        struct mallinfo malloc_info = mallinfo(); // structure has same name as function that returns it. (see malloc.h)
```
Comment (Soullivaneuh) [inline src/mongo/util/processinfo_linux2.cpp]: TODO: Restore it.

**Comment by Soullivaneuh** [comment]
Can you please tell me which libC do you use for compilation?

I'm quite surprice because even on Ubuntu Glibc, I don't have some functions, like `walkcontext()` for examples.

```bash
$ ldd --version
ldd (Ubuntu GLIBC 2.21-0ubuntu4) 2.21
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

**Comment by franz-josef-kaiser** [comment]
@Soullivaneuh I _guess_ the only missing piece for traction here is a Jira–Issue [in the MongoDB issue tracker here](https://jira.mongodb.org/secure/Dashboard.jspa).

**Comment by Soullivaneuh** [comment]
Unfortunately, I don't have more allowed work time for this implementation right now.

Feel free to cherry-pick and continue my PR. I also gave maintainers edit here.

Regards.

**Comment by Schubes** [comment]
As MongoDB 2.6 is no longer supported, I'm going to close this pull request in favor of #1061.

**Pull request closed without merging**